OAK-Marketplace-Emotion-Recognition/emotion_analyzer.py

```diff
@@ -0,0 +1,257 @@
+        top_bar.update_bar(top_bar_progress)
+        top_bar_x = int(self.result_bar_size / 2 - top_bar.width / 2)
+        top_bar_y = int(self.result_bar_size / 2 - top_bar.height / 2)
+        text = f'MOSTLY {top_bar.emotion.upper()} - {self.get_current_emotion_percent(top_bar.emotion)}%'
+        result_statistic = add_class_names_and_percents(result_statistic,
+                                                          [top_bar_x,
+                                                           int(top_bar_y - 2 * Y_OFFSET / 2)],
+                                                           text)
+        overlay_image_alpha(result_statistic, top_bar.bar, top_bar_x, top_bar_y, top_bar.alpha_mask)
+        start_x, start_y = np.clip(result_statistic.shape[0] - 2 * self.bars_size, 0, result_statistic.shape[0]) , \
+                               np.clip(result_statistic.shape[1] - self.bars_size - Y_OFFSET, 0, result_statistic.shape[0])
+        for emotion, emotion_bar in self.bars.items():
+            if emotion == top_emotion:
+                continue
+            percent = self.get_current_emotion_percent(emotion)
+            text = f'{emotion.upper()} - {percent}%'
+            result_statistic = add_class_names_and_percents(result_statistic, [start_x, int(result_statistic.shape[1] - Y_OFFSET / 2)],
+                                                        text)
+            alpha_mask = emotion_bar.alpha_mask
+            overlay_image_alpha(result_statistic, emotion_bar.bar, start_x, start_y, alpha_mask)
+            start_x = np.clip(start_x - self.bars_size - 15 * X_OFFSET, 0, result_statistic.shape[0])
+        if save:
+            cv2.imwrite('result_statistic_emotion_bar.png', result_statistic)
+        return  result_statistic
+
+
+class EmotionBar:
+    def __init__(
+        self, background_color: tuple = (28, 28, 30),
+        progress_color: tuple = (145, 82, 225),
+        size: int = 80,
+        thickness: int = 10,
+        start_angle: int = -220,
+        end_angle: int = 40,
+        emotion: str = 'happy',
+        emotion_amount: int = 0,
+        progress: float = 0.0,
+        emotion_size: tuple = (25, 25),
+        emotions_path: str = 'images/emoji',
+    ):
+        self.emotion_images = {
+            'neutral': np.array(
+                Image.open(os.path.join(emotions_path, "neutral.png")).resize(emotion_size)),
+            'happy': np.array(
+                Image.open(os.path.join(emotions_path, "happy.png")).resize(emotion_size),
+            ),
+            'sad': np.array(
+                Image.open(os.path.join(emotions_path, "sad.png")).resize(emotion_size)),
+            'anger': np.array(
+                Image.open(os.path.join(emotions_path, "anger.png")).resize(emotion_size),
+            ),
+            'surprise': np.array(
+                Image.open(os.path.join(emotions_path, "surprise.png")).resize(emotion_size)),
```

Comment:
Please, don't hardcode emotions. EmotionBar can be any emotion. Not only those ones. Please, move use it as a parameter like emotion_path